### PR TITLE
[DX] Help developers choose the right branch

### PR DESCRIPTION
**Pull request by SpacePossum**
Simular to: http://symfony.com/doc/current/contributing/code/patches.html#choose-the-right-branch
Base on feedback here: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1128 and

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
No point in this. This will only get outdated. The 1.5 branch would have been deleted if it weren't for the fact that github won't let us because there are pulls targeted at it.

**Comment by GrahamCampbell** [comment]
@SpacePossum It's not up to me to reject this. You should leave this for @keradus to decide.

**Comment by SpacePossum** [comment]
ah ok :+1: thanks!

**Comment by keradus** [comment]
Don't like putting version exactly... they will change after each minor releases...
current default branch for features, previous branch for bugfixes, master branch for BC breaks

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -3,7 +3,10 @@
 ## Quick guide
 
  * Fork the repo.
- * Checkout the branch you want to make changes on.
+ * Checkout the branch you want to make changes on:
+  * Previous release branch if you are fixing a bug for an existing feature (or the current release branch if the feature you are fixing was introduced in that version)
```
Comment (keradus) [inline CONTRIBUTING.md]: missing punctuation marks on the end of sublist's items

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -3,7 +3,10 @@
...
+  * Current release branch if you are adding a new feature
+  * Master if you are adding a new feature or make changes to the code that are not backward compatible
```
Comment (keradus) [inline CONTRIBUTING.md]: so... for new feature we could go in both current release and master ? master is for BC breakers

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -3,7 +3,10 @@
 ## Quick guide
 
  * Fork the repo.
- * Checkout the branch you want to make changes on.
+ * Checkout the branch you want to make changes on:
+  * Previous release branch if you are fixing a bug for an existing feature (or the current release branch if the feature you are fixing was introduced in that version)
+  * Current release branch if you are adding a new feature
```
Comment (keradus) [inline CONTRIBUTING.md]: what is current release branch, why not default branch ?

File: CONTRIBUTING.md (hunk 1 above)
Comment (keradus) [inline CONTRIBUTING.md]: Descr for prev branch and then... current branch. And what if bug was made on master?

What about describing it another way:
- master for BC breakers
- current branch for new features
- ...

?

**Comment by keradus** [comment]
:+1:

**Comment by gharlan** [comment]
:+1:

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request closed without merging**